src/core/public/chrome/ui/header/branding/opensearch_dashboards_custom_logo.tsx

```diff
@@ -35,23 +35,29 @@ import '../header_logo.scss';
 import { OpenSearchDashboardsLogoDarkMode } from './opensearch_dashboards_logo_darkmode';
 
 /**
- * @param {string} logoUrl - custom URL for the top left logo of the main screen
+ * @param {string} fullLogoUrl - custom URL for the top left logo of the main screen
+ * @param {string} logoUrl - custom URL for the logo icon
+ * @param {string} title - custom title for the application
  */
 export interface CustomLogoType {
+  fullLogoUrl?: string;
   logoUrl?: string;
+  title: string;
 }
 
 export const CustomLogo = ({ ...branding }: CustomLogoType) => {
-  return !branding.logoUrl ? (
-    OpenSearchDashboardsLogoDarkMode()
-  ) : (
-    <img
-      data-test-subj="customLogo"
-      data-test-image-url={branding.logoUrl}
-      src={branding.logoUrl}
-      alt="logo"
-      loading="lazy"
-      className="logoImage"
-    />
-  );
+  if (!branding.fullLogoUrl && !branding.logoUrl) {
+    return OpenSearchDashboardsLogoDarkMode();
+  } else {
+    return (
+      <img
+        data-test-subj="customLogo"
+        data-test-image-url={!branding.fullLogoUrl ? branding.logoUrl : branding.fullLogoUrl}
```

Comment:
line 55 and 56 looks like dup we can potentially pull it out and ignore the NIT comment above.

Reply:
Pulled out to reduce repetition: const headerLogoUrl = !branding.fullLogoUrl ? branding.logoUrl : branding.fullLogoUrl.